lib/client-options.js

```diff
@@ -113,6 +115,10 @@ function extend(baseOptions, userOptions) {
     throw new TypeError('requestTracker must be an instance of RequestTracker');
   }
 
+  if (!(options.metrics instanceof metrics.ClientMetrics)) {
+    throw new TypeError('metrics must be an instance of ClientMetrics or null');
```

Comment:
technically, wouldn't it being null be an issue as we invoke the `onXXX` methods without null checking?  Should it be strictly non-null or a `ClientMetrics` instance?

Reply:
Good catch, initially I was setting the default instance later but I changed the behaviour and not the error message.
I'll change it.